File: pkg/models/session.go
package models

type Session struct {
	Id              string `json:"id"`
	StartTime       int64  `json:"startTime"`
	SessionType     string `json:"sessionType"`
	Track           string `json:"track"`
	CarModel        string `json:"carModel"`
	NumberOfSectors int32  `json:"numberOfSectors"`
	LapsCompleted   int32  `json:"lapsCompleted"`
	BestLap         int32  `json:"bestLap"`
	IsActive        bool   `json:"isActive"`
	Player          string `json:"player"`

	Laps []*Lap `json:"laps"`
}

type Lap struct {
	LapNumber int32 `json:"lapNumber"`
	LapTime   int32 `json:"lapTime"`
	LapDelta  int32 `json:"lapDelta"`
	IsValid   bool  `json:"isValid"`
	IsActive  bool  `json:"isActive"`

	LapSectors []*LapSector `json:"lapSectors"`
}

type LapSector struct {
	SectorNumber int32 `json:"sectorNumber"`
	SectorTime   int32 `json:"sectorTime"`
	IsActive     bool  `json:"isActive"`
}

File: web/models/session.ts
export interface LapSector {
  sectorNumber: number;
  sectorTime: number;
  isActive: boolean;
}

export interface Lap {
  lapNumber: number;
  lapTime: number;
  isValid: boolean;
  isActive: boolean;
  lapSectors: LapSector[];
}

export interface Session {
  id: string;
  startTime: number;
  sessionType: string;
  track: string;
  carModel: string;
  numberOfSectors: number;
  completedLaps: number;
  bestLapTime: number;
  previousLapTime: number;
  isActive: boolean;
  player: string;
  laps: Lap[];
}
